technical_architecture/performances_guide/batch_page_size.rst

```diff
@@ -0,0 +1,13 @@
+Job product batch size
+======================
+
+A product catalog is unique to one's need. It is configured with a custom number of attributes, families, locales, channels, etc. Those different combinations may affect the import/exports performances.
+
+Indeed the size of the products will define how much RAM they will take to be processed. During product import/exports, we regularly flush the processed products into the database to avoid too much memory consumption.
+
+But, if we flush them too often, the transaction time can slow the process: flushing is not free and doing it too often will be inefficient. On the other hand, if we flush too sporadically, the RAM consumption can increase to a point where the PHP's garbage collector is called too many times and slows down the import/exports.
```

Comment:
Actually, it's not the transaction time that is the bottleneck (and so, flushing). However, Doctrine probably adds an overhead.

We batch to have the benefit of not loading everything everytime from the DB.
We clear because the avilable memory is limited and we can't load the whole database in memory.

Reply:
We can add that during at the end of the batch, we index the batch of product in Elasticsearch. If we reduce the batch size, we index more often. And the cost of communication between PHP and Elasticsearch begins to cumulate, and being counter productive.